treeherder/intermittents_commenter/management/commands/run_intermittents_commenter.py

```diff
@@ -1,16 +1,27 @@
+import calendar
+from datetime import date
+
 from django.core.management.base import BaseCommand
 
 from treeherder.intermittents_commenter.commenter import Commenter
 
 
 class Command(BaseCommand):
     """Management command to manually initiate intermittent failures commenter
-       default is daily and non-test mode, use flags for weekly and test mode."""
+       default is daily and non-test mode, use flags for weekly, auto and test mode."""
 
     def add_arguments(self, parser):
-        parser.add_argument('-w', '--weekly', action='store_true', dest='weekly_mode', help='generate weekly, rather than daily, comment summaries')
+        parser.add_argument('-m', '--mode', dest='weekly_mode', nargs='?', choices=['weekly', 'auto'], default=False, help='generate comment summaries based on auto or weekly mode')
         parser.add_argument('--dry-run', action='store_true', dest='dry_run', help='output comments to stdout rather than submitting to Bugzilla')
 
     def handle(self, *args, **options):
-        process = Commenter(weekly_mode=options['weekly_mode'], dry_run=options['dry_run'])
+        mode = options['weekly_mode']
+        weekly_mode = mode
```

Comment:
This was a bit confusing since at first glance it reads roughly like:

```
x = y
y = x
```

What do you think about:
* storing the `--mode` value in `mode` rather than `weekly_mode` (ie `dest='mode'` in `add_arguments`)
* replacing the `weekly_mode = mode` line with:

```python
is_monday = calendar.day_name[date.today().weekday()] == 'Monday'
weekly_mode = (mode == 'weekly') or (mode == 'auto' and is_monday)
```